kernel/src/scan/data_skipping.rs

```diff
@@ -304,83 +182,100 @@ impl DataSkippingFilter {
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+struct DataSkippingPredicateCreator;
+
+impl DataSkippingPredicateEvaluator for DataSkippingPredicateCreator {
+    type Output = Expr;
+    type TypedStat = Expr;
+    type IntStat = Expr;
+
+    /// Retrieves the minimum value of a column, if it exists and has the requested type.
+    fn get_min_stat(&self, col: &ColumnName, _data_type: &DataType) -> Option<Expr> {
+        Some(joined_column_expr!("minValues", col))
```

Comment:
nit: should probably pull out `"minValues"` etc. as constants?

Reply:
If we make them constants, we can't use the macro any more and would end up with e.g.
```suggestion
        Some(ColumnName::new([MIN_VALUES_COL_NAME]).join(col))
```
... which is certainly doable, but also a bit of a mouthful.

Also, there used to be more uses of these magic constants, but this PR leaves only two prod uses of each of those constants: One for its `get_xxx_stat` method here, and the `stats_schema` created internally by `DataSkippingFilter::new`. 

Thoughts?

Reply:
hm fair point. this seems like not worth spending time on so let's leave as-is. next time someone wants to play with macros we could probably expand the `column_name!` etc. to accept a `path` in addition to a string literal? but again doesn't really feel worth the time right now :)

Reply:
How would a `path` help? The problem is, macros run before const evaluates (indeed, before we even know what _type_ the const evaluates to), so the macro couldn't verify the content of a const string meets the safety conditions.